### DISCUSSION TITLE
Practical usage of useFetch’s getCachedData

### DISCUSSION THREAD

**Discussion by hirotaka** [nuxt/nuxt#24299] [Questions]
When developing an application and the scale of the application increases, the parts of the application that request data from the API become more complex.

So, I'm trying to see if I can improve it using `getCachedData` introduced in Nuxt 3.8 ([YouTube Video](https://www.youtube.com/watch?v=aQPR0xn-MMk) was accommodating. Thank you very much!).

I want to make a single request to the API, even if I do the fetch process from several components, like `@tanstack/query`.

I've accomplished most of my goals with the following repository, but I'm eager to continue improving and discovering better approaches.

[![Open in
  StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github.com/hirotaka/mre/tree/main/nuxt-use-fetch-get-cached-data)

Questions:

First, I implemented to return cached data for 5 seconds after the first request to prevent duplicate requests. It works fine in most cases, but I want to improve the process. For example, return cached data as long as the fetch process is called from the same page.

Also, if `useFetch` runs simultaneously, the fetch will run twice. For example, `LikeIcon` and `LikeButton`. To prevent this, I've added a sleep. Is there any smarter way to archive this?

If you notice anything else, I would be glad to hear.

Thanks.

**Comment by Amar-Gill** [comment]
Can consider using `useNuxtData(key)` instead for a situation where two concurrent requests are happening. And then add a watcher for the cached data when it changes from null to not null. That's what I'm doing in a project right now.
